core/include/cubos/core/gl/light.hpp

```diff
@@ -0,0 +1,44 @@
+
+#ifndef CUBOS_CORE_GL_LIGHT_DATA_HPP
+#define CUBOS_CORE_GL_LIGHT_DATA_HPP
+
+#include <glm/glm.hpp>
+#include <glm/gtc/quaternion.hpp>
+
+namespace cubos::core::gl
+{
+    struct SpotLightData
```

Comment:
I think we can rename these structs to `SpotLight` etc (remove the data) since the other 'data' types do not end in 'data' (`Vertex`, `Grid`, `Palette`).